diff --git a/test/e2e/brand_customization_test.go b/test/e2e/brand_customization_test.go
index 053c56589..6f9eee48b 100644
--- a/test/e2e/brand_customization_test.go
+++ b/test/e2e/brand_customization_test.go
@@ -23,6 +23,7 @@ import (
 	"github.com/openshift/console-operator/test/e2e/framework"
 )

+// TODO remove deprecated CustomLogoFile API
 const (
 	customLogoVolumeName       = "custom-logo"
 	customLogoMountPath        = "/var/logo/"
@@ -189,8 +190,11 @@ func hasCustomBranding(cm *v1.ConfigMap, desiredProductName string, desiredLogoF
 	consoleConfig := consoleserver.Config{}
 	yaml.Unmarshal([]byte(cm.Data["console-config.yaml"]), &consoleConfig)
 	actualProductName := consoleConfig.Customization.CustomProductName
-	actualLogoFile := consoleConfig.Customization.CustomLogoFile
-	return (desiredProductName == actualProductName) && ("/var/logo/"+desiredLogoFileName == actualLogoFile)
+	// TODO remove deprecated CustomLogoFile API
+	// Custom logo file used for both dark and light themes
+	actualLogoFile1 := consoleConfig.Customization.Logos[0].Themes[0].Source.ConfigMap.Name
+	actualLogoFile2 := consoleConfig.Customization.Logos[0].Themes[1].Source.ConfigMap.Name
+	return (desiredProductName == actualProductName) && (desiredLogoFileName == actualLogoFile1) && (desiredLogoFileName == actualLogoFile2)
 }

 func findCustomLogoVolume(deployment *appsv1.Deployment) bool {
